TriggerRedHatBuild.groovy

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2020-present Sonatype, Inc. All rights reserved.
+ * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
+ * "Sonatype" is a trademark of Sonatype, Inc.
+ */
+@Grab('io.github.http-builder-ng:http-builder-ng-core:1.0.4')
+
+import groovyx.net.http.HttpBuilder
+import groovyx.net.http.HttpException
+
+if (args.size() < 3) {
+  println 'Usage: groovy TriggerRedhatBuild.groovy <version> <projectId> <apiKey>'
+  return
+}
+
+def (version, projectId, apiKey) = args
+
+final nextTag = getNextTag(apiKey, projectId, version)
+println "Deploying as ${nextTag}"
+
+build(apiKey, projectId, nextTag)
+
+final completedBuild = getCompletedBuild(apiKey, projectId, nextTag)
+
+println publish(apiKey, projectId, completedBuild.digest, completedBuild.name)
+
+String getNextTag(String apiKey, String projectId, String version) {
+  final tags = HttpBuilder.configure {
+    request.uri = "https://connect.redhat.com/api/v2/projects/${projectId}/tags"
+    request.headers['Authorization'] = "Bearer ${apiKey}"
+    request.contentType = 'application/json'
+    request.body = [:]
+  }.post().tags*.name.collectMany {
+    it.split(', ').collect()
+  }
+
+  final currentIndex = tags.findAll {
+    it.startsWith(version)
+  }.collect {
+    it.replaceAll(/${version}-(\d+)-?.*/, '$1') as Integer
+  }.sort().reverse()[0]
+
+  final nextIndex =((currentIndex ?: 0) as Integer) + 1
+
+  return "${version}-${nextIndex}"
+}
+
+Map build(String apiKey, String projectId, String nextTag) {
+  return HttpBuilder.configure {
+    request.uri = "https://connect.redhat.com/api/v2/projects/${projectId}/build"
+    request.headers['Authorization'] = "Bearer ${apiKey}"
+    request.contentType = 'application/json'
+    request.body = [tag: nextTag]
+  }.post()
+}
+
+Map getCompletedBuild(String apiKey, String projectId, String nextTag) {
+  while (true) {
+    println "Waiting for build to finish."
+    sleep 60000
+
+    final newTags = HttpBuilder.configure {
+      request.uri = "https://connect.redhat.com/api/v2/projects/${projectId}/tags"
+      request.headers['Authorization'] = "Bearer ${apiKey}"
+      request.contentType = 'application/json'
+      request.body = [:]
+    }.post().tags
+
+    final completedBuild = newTags.find {
+      it.name == nextTag && it.scan_status == 'passed'
+    }
+
+    if (completedBuild) {
+      return completedBuild
+    }
+  }
+}
+
+Map publish(String apiKey, String projectId, String digest, String name) {
+  final publishUri = [
+    "https://connect.redhat.com/api/v2/projects",
+    projectId,
+    "containers",
+    digest,
+    "tags",
+    name,
+    "publish"
+  ].join('/')
+
+  try {
+    return HttpBuilder.configure {
```

Comment:
what does HttpBuilder provide that you can't do with HttpUrlConnection? This way you could have 0 dependencies.

Reply:
for example like this: https://github.com/sonatype/iq-jira-plugin/blob/master/testsuite/technical-testsuite/src/test/java/com/sonatype/jira/iq/ui/ConfigurationPageIT.groovy#L180
but up to you what you prefer.

Reply:
The built-in json parsing is convenient, and this looks nice and concise in my opinion. This library seemed popular in google searches for groovy http libraries. I can switch to the HttpUrlConnection and my own json slurping if you guys insist. It looks like you're not insisting though.

Reply:
I might also be able to refactor this to reuse the common parts and push the variations into the `post` closure. 🤔

Reply:
no, i don't insist at all, whatever you prefer to do.

Reply:
Flip side - why NOT use this library? ;)

Reply:
I'm about to push a nice little cleanup to reuse the builder to reduce repetition of some of the config. It's a nice library.